bigquery/storage/managedwriter/adapt/protoconversion.go

```diff
@@ -286,46 +287,73 @@ func storageSchemaToDescriptorInternal(inSchema *storagepb.TableSchema, scope st
-		outType := bqTypeToFieldTypeMap[field.GetType()]
-		fdp := &descriptorpb.FieldDescriptorProto{
-			Name:   proto.String(name),
-			Number: proto.Int32(idx),
-			Type:   outType.Enum(),
-			Label:  convertModeToLabel(field.GetMode(), useProto3),
 		}
-		// Special case: proto2 repeated fields may benefit from using packed annotation.
-		if field.GetMode() == storagepb.TableFieldSchema_REPEATED && !useProto3 {
-			for _, v := range packedTypes {
-				if outType == v {
-					fdp.Options = &descriptorpb.FieldOptions{
-						Packed: proto.Bool(true),
+	} else {
+		// For (REQUIRED||REPEATED) fields for proto3, or all cases for proto2, we can use the expected scalar types.
+		if field.GetMode() != storagepb.TableFieldSchema_NULLABLE || !useProto3 {
+			outType := bqTypeToFieldTypeMap[field.GetType()]
+			fdp = &descriptorpb.FieldDescriptorProto{
+				Name:   proto.String(name),
+				Number: proto.Int32(idx),
+				Type:   outType.Enum(),
+				Label:  convertModeToLabel(field.GetMode(), useProto3),
+			}
+
+			// Special case: proto2 repeated fields may benefit from using packed annotation.
+			if field.GetMode() == storagepb.TableFieldSchema_REPEATED && !useProto3 {
+				for _, v := range packedTypes {
+					if outType == v {
+						fdp.Options = &descriptorpb.FieldOptions{
+							Packed: proto.Bool(true),
+						}
+						break
 					}
-					break
 				}
 			}
+		} else {
+			// For NULLABLE proto3 fields, use a wrapper type.
+			fdp = &descriptorpb.FieldDescriptorProto{
+				Name:     proto.String(name),
+				Number:   proto.Int32(idx),
+				Type:     descriptorpb.FieldDescriptorProto_TYPE_MESSAGE.Enum(),
+				TypeName: proto.String(bqTypeToWrapperMap[field.GetType()]),
+				Label:    descriptorpb.FieldDescriptorProto_LABEL_OPTIONAL.Enum(),
+			}
 		}
-		return fdp, nil
 	}
-	// For NULLABLE proto3 fields, use a wrapper type.
-	return &descriptorpb.FieldDescriptorProto{
-		Name:     proto.String(name),
-		Number:   proto.Int32(idx),
-		Type:     descriptorpb.FieldDescriptorProto_TYPE_MESSAGE.Enum(),
-		TypeName: proto.String(bqTypeToWrapperMap[field.GetType()]),
-		Label:    descriptorpb.FieldDescriptorProto_LABEL_OPTIONAL.Enum(),
-	}, nil
+	if nameRequiresAnnotation(name) {
+		// Use a prefix + base64 encoded name when annotations bear the actual name.
+		// Base 64 standard encoding may also contain certain characters (+,/,=) which
+		// we remove from the generated name.
+		encoded := strings.Trim(base64.StdEncoding.EncodeToString([]byte(name)), "+/=")
+		fdp.Name = proto.String(fmt.Sprintf("col_%s", encoded))
+		opts := fdp.GetOptions()
+		if opts == nil {
+			fdp.Options = &descriptorpb.FieldOptions{}
+		}
+		proto.SetExtension(fdp.Options, storagepb.E_ColumnName, name)
+	}
+	return fdp, nil
+}
+
+// nameRequiresAnnotation determines whether a field name requires unicode-annotation.
+func nameRequiresAnnotation(in string) bool {
+	for i := 0; i < len(in); i++ {
+		if in[i] > unicode.MaxASCII {
```

Comment:
proto does not only refuse to take unicode, it refuses to take anything outside of the alphnumeric+underscore character set. it also has restrictions on initial characters, i believe digits are not allowed as first character. 

So we should expand the check here to have more names put in the annotation.

Reply:
Went looking for a better source of truth, found it in protoreflect and swapped the check for that.